components/proxy/src/main/java/com/hotels/styx/metrics/reporting/graphite/GraphiteReporter.java

```diff
@@ -348,12 +360,18 @@ private void doSend(String name, double value, long timestamp) {
         doSend(name, format(value), timestamp);
     }
 
+
     private void doSend(String name, String value, long timestamp) {
-        try {
-            graphite.send(name, value, timestamp);
-        } catch (Exception e) {
-            reconnectIfNecessary(e);
-        }
+        attemptWithRetryAndReconect(
+                () -> graphite.send(name, value, timestamp));
+    }
+
+    private void attemptWithRetryAndReconect(IoRetry.IoRunnable operation) {
+        IoRetry.tryTimes(
+                MAX_RETRIES,
+                operation,
+                this::reconnectIfNecessary
+        );
```

Comment:
It looks to me that this approach will not address the issue. This is because `doSend` and `attemptWithRetryAndReconnect` are called once for each metric, resetting the retry count to `MAX_RETRIES` for each metric. As a result it will end up attempting each metric `MAX_RETRIES` times. 

Instead the `MAX_RETRIES` should be set to each reporting cycle such that the report function tolerates maximum of `MAX_RETRIES` before giving up.